src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiImporter.cs

```diff
@@ -0,0 +1,370 @@
+				javaName: element.XGetAttribute ("name"),
+				javaVisibility: element.XGetAttribute ("visibility"),
+				javaAbstract: element.XGetAttributeAsBool ("abstract"),
+				javaFinal: element.XGetAttributeAsBool ("final"),
+				javaStatic: element.XGetAttributeAsBool ("static"),
+				javaParentType: type,
+				deprecated: element.XGetAttribute ("deprecated"),
+				jniSignature: element.XGetAttribute ("jni-signature"),
+				isSynthetic: element.XGetAttributeAsBool ("synthetic"),
+				isBridge: element.XGetAttributeAsBool ("bridge")
+			);
+
+			if (element.Element ("typeParameters") is XElement tp)
+				ParseTypeParameters (method.TypeParameters, tp);
+			foreach (var child in element.Elements ("exception"))
+				method.Exceptions.Add (ParseException (child));
+
+			foreach (var child in element.Elements ("parameter"))
+				method.Parameters.Add (ParseParameter (method, child));
+
+			if (element.XGetAttribute ("merge.SourceFile") is string source && source.HasValue ())
+				method.PropertyBag.Add ("merge.SourceFile", source);
+			if (element.XGetAttribute ("deprecated-since") is string dep && dep.HasValue ())
+				method.PropertyBag.Add ("deprecated-since", dep);
+
+			return method;
+		}
+
+		public static JavaFieldModel ParseField (JavaTypeModel type, XElement element)
+		{
+			var field = new JavaFieldModel (
+				name: element.XGetAttribute ("name"),
+				visibility: element.XGetAttribute ("visibility"),
+				type: element.XGetAttribute ("type"),
+				typeGeneric: element.XGetAttribute ("type-generic-aware"),
+				isStatic: element.XGetAttributeAsBool ("static"),
+				value: element.Attribute ("value")?.Value,
+				parent: type,
+				isFinal: element.XGetAttributeAsBool ("final"),
+				deprecated: element.XGetAttribute ("deprecated"),
+				jniSignature: element.XGetAttribute ("jni-signature"),
+				isTransient: element.XGetAttributeAsBool ("transient"),
+				isVolatile: element.XGetAttributeAsBool ("volatile"),
+				isNotNull: element.XGetAttributeAsBool ("not-null")
+			);
+
+			if (element.XGetAttribute ("merge.SourceFile") is string source && source.HasValue ())
+				field.PropertyBag.Add ("merge.SourceFile", source);
+			if (element.XGetAttribute ("deprecated-since") is string dep && dep.HasValue ())
+				field.PropertyBag.Add ("deprecated-since", dep);
+
+			return field;
+		}
+
+		public static JavaImplementsModel ParseImplements (XElement element)
+		{
+			var model = new JavaImplementsModel (
+				name: element.XGetAttribute ("name"),
+				nameGeneric: element.XGetAttribute ("name-generic-aware"),
+				jniType: element.XGetAttribute ("jni-type")
+			);
+
+			if (element.XGetAttribute ("merge.SourceFile") is string source && source.HasValue ())
+				model.PropertyBag.Add ("merge.SourceFile", source);
+
+			return model;
+		}
+
+		public static JavaExceptionModel ParseException (XElement element)
+		{
+			return new JavaExceptionModel (
+				name: element.XGetAttribute ("name"),
+				type: element.XGetAttribute ("type-generic-aware")
```

Comment:
While reviewing `api-30.xml.in`, I see that we didn't quite get `jni-*` attributes everywhere; `//exception/@name` is just the type name from `//exception/@type`, which feels a tad redundant.  `//exception/@type`, meanwhile, is a *Java* type name, e.g. [`Resources.NotFoundException`](https://developer.android.com/reference/android/content/res/Resources.NotFoundException) is represented as:

```xml
<exception name="Resources$NotFoundException" type="android.content.res.Resources.NotFoundException" />
```

which is *not* an unambiguous format, though I suppose we could `@type.Substring(0, @type.IndexOf(@name))` to extract the package name, if necessary.

We should probably update the XML to instead be:

```xml
<exception name="Resources$NotFoundException" type="android.content.res.Resources.NotFoundException" jni-type="android/content/res/Resources$NotFoundException" />
```